src/factory.ts

```diff
@@ -83,7 +83,20 @@ export class Factory {
   /** To enable logging for this class. Set `Vex.Flow.Factory.DEBUG` to `true`. */
   static DEBUG: boolean;
 
-  protected options: FactoryOptions;
+  /**
+   * Static simplified function to access constructor without providing FactoryOptions
+   *
+   * Example:
+   *
+   * Create an SVG renderer and attach it to the DIV element named "boo" to render using <page-width> 1200 and <page-height> 600
+   *
+   * `const vf: Factory = Vex.Flow.Factory.newFromElementId('boo', 1200, 600 );`
+   */
+  static newFromElementId(elementId: string | null, width = 500, height = 200): Factory {
+    return new Factory({ renderer: { elementId, width, height, backend: Renderer.Backends.SVG } });
+  }
+
+  protected options: Required<FactoryOptions>;
```

Comment:
While we're at it, maybe we shouldn't be calling required parameters "options". The constructor should have explicit parameters for required parameters, and an options struct for optional stuff. Maybe here, we'll call it `params`?

Reply:
`Required<...>` is an annotation that tells the compiler (and someone reading the code) that we know all the fields are set to something (i.e., not undefined).

Should we rename all fields that are called `options` that we know have no undefined fields? I'm happy to call them all `params` as long as we are consistent.

Reply:
But what about accessors? Should they be getOptions() / setOptions(), or getParams() / setParams(), or getParams() / setOptions() ??? :-)

Reply:
I'd argue that, in the general case, we shouldn't have accessors for `options` or `params`.

Anyhow, this isn't a strong opinion -- if this were new code, I'd be more firm about separating required parameters from options, but since we're porting JS code, it's okay.

Reply:
I did some thinking and also read some (only slightly) related posts:
- https://stackoverflow.com/questions/36495669/difference-between-terms-option-argument-and-parameter/36495940
- https://stackoverflow.com/questions/12826977/multiple-arguments-vs-options-object
- https://www.reddit.com/r/typescript/comments/7t8jpw/creating_a_class_with_options_object_instead_of/

It seems like "**Options Object**" is a commonly understood name for the _incoming thing_ that is provided to the constructor. 

However, **after we validate these options**, should we keep them inside a `this.options` field that is typed as `Required<XXXOptions>`?

I think the most "correct" way would be to not keep an internal options object at all. We would validate all the incoming options, and then assign them directly to instance variables which determine how the element will be rendered.

After the constructor has done its work, all the instance variables will have valid values, and we can use them directly:

- `this.options.fill_style` => `this.fillStyle`
- `this.options.stave.space` => `this.staveSpace`
- ...

There are 8 instances where we call `stave.getOptions()`. Other elements rely on the stave's internal metrics. In the world where we abandon internal options objects, all of those would instead call the individual readonly accessor for each field:
- `stave.getOptions().vertical_bar_width` => `stave.getVerticalBarWidth()`
- `stave.getOptions().spacing_between_lines_px` => `stave.getSpacingBetweenLines()`
- `stave.getOptions().num_lines` => `stave.getNumLines()`


I kind of like that, to be honest!

Do you think we should do that for the 4.0 release? Or should we just try it on `stave.ts` only and see what we think?

Reply:
OK, I made a slight improvement here.

There were only two declarations of `getOptions()`. Stave and Factory. The Factory one was only used by a test case (sanity check). So I removed it and accessed the protected member directly. The Stave.getOptions() was also unnecessary. Most of the options already had accessors. I implemented one additional accessor: stave.getVerticalBarWidth().

So now there are no more accessors for options.

We can discuss the other ideas (e.g., whether there should even be an internal `this.options` object) post 4.0 release, since it should only affect the internals of each class.